app/javascript/lib/study-overview.js

```diff
@@ -0,0 +1,338 @@
+      x: aspects.x,
+      y: aspects.y,
+      z: aspects.z
+    }
+  }
+
+  return scene
+}
+
+/** Gets Plotly layout props for 2D scatter plot */
+export function get2DScatterProps(cluster) {
+  const {
+    axes, clusterHasRange, domainRanges, hasCoordinateLabels, coordinateLabels
+  } = cluster
+  const { titles } = axes
+
+  const layout = {
+    xaxis: { title: titles.x, showticklabels: false },
+    yaxis: { title: titles.y, showticklabels: false, scaleanchor: 'x' }
+  }
+
+  // if user has supplied a range, set that, otherwise let Plotly autorange
+  if (clusterHasRange) {
+    layout.xaxis.range = domainRanges.x
+    layout.yaxis.range = domainRanges.y
+  } else {
+    layout.xaxis.autorange = true
+    layout.yaxis.autorange = true
+  }
+
+  if (hasCoordinateLabels) {
+    layout.annotations = coordinateLabels
+  }
+
+  return layout
+}
+
+/** Get height and width for a to-be-rendered cluster plot */
+function calculatePlotRect() {
+  const numPlots = window.SCP.numPlots
+
+  const height = $(window).height() - 250
+  const width = ($('#plots-tab').width() - 80) / numPlots
+
+  return { height, width }
+}
+
+
+/** Resize Plotly scatter plots -- done on window resize  */
+function resizePlots() {
+  const numPlots = window.SCP.numPlots
+
+  for (let i = 0; i < numPlots; i++) {
+    const rawPlot = window.SCP.plots[i]
+    const layout = getScatterPlotLayout(rawPlot)
+    const target = `cluster-plot-${i + 1}`
+
+    Plotly.relayout(target, layout)
+  }
+}
+
+
+/** Renders Plotly scatter plot for "Clusters" tab */
+function renderScatterPlot(target, rawPlot) {
+  const { data } = rawPlot
+
+  window.SCP.scatterCount += 1
+  const scatterCount = window.SCP.scatterCount
+
+  const plotId = `cluster-plot-${scatterCount}`
+
+  $(target).append(`
+    <div class="row" style="float: left">
+      <div id="${plotId}"></div>
+      <div id="cluster-figure-legend"></div>
+    </div>`)
+
+  const layout = getScatterPlotLayout(rawPlot)
+
+  Plotly.newPlot(plotId, data, layout)
+
+  // listener to redraw expression scatter with new color profile
+  $('#colorscale').off('change')
+  $('#colorscale').change(function() {
+    const theme = $(this).val() // eslint-disable-line
+    data[0].marker.colorscale = theme
+    console.log(`setting colorscale to ${theme}`)
+
+    $('#search_colorscale').val(theme)
+    Plotly.update(plotId, data, layout)
+  })
+
+  const description =
+    `<p class="text-center help-block">${window.SCP.cluster.description}</p>`
+  $('#cluster-figure-legend').html(description)
+
+  // access actual target div, not jQuery object wrapper for relayout event
+  const clusterPlotDiv = document.getElementById(plotId)
+  clusterPlotDiv.on('plotly_relayout', cameraData => {
+    if (typeof cameraData['scene.camera'] !== 'undefined') {
+      const newCamera = cameraData['scene.camera']
+      $(`#${plotId}`).data('camera', newCamera)
+    }
+  })
+
+  window.SCP.scatterPlotLayout = layout
+}
+
+/**
+ * End render cluster code
+ */
+
```

Comment:
when I see comments like the above, I think 'I wonder if that means the stuff above/below should be in different files?'

Reply:
Same, that's my intent.  See other comment about just-ticketed work.